builder/frameworks/arduino/stm32duino.py

```diff
@@ -26,7 +26,35 @@
 
 env = DefaultEnvironment()
 
-if "stm32f1" in env.BoardConfig().get("build.variant"):
+# supported mcu types by official STM32 Arduino core
+supported = [
+    "f030r8", "f091rc",
+    "f100rb",
+    "f103c", "f103rb",
+    "f207zg",
+    "f302r8", "f303k8", "f303re",
+    "f401re", "f411re", "f429zi", "f446re", "f407vg",
+    "f746ng",
+    "l031k6", "l053r8", "l072cz",
+    "l152re",
+    "l432kc", "l476rg", "l496zg", "l475vg"
+]
+
+mcu_type = env.BoardConfig().get("build.mcu")
+official = False
+
+if not "f103" in mcu_type and not "f407" in mcu_type:
+    official = True
+
+if "STM32_OFFICIAL_CORE" in env['CPPDEFINES']:
```

Comment:
Could you replace this macro with `PIO_FRAMEWORK_ARDUINO_OFFICIAL` to make it clear that it is related just to PlatformIO?